apple/RNCWebView.h

```diff
@@ -101,18 +101,24 @@ shouldStartLoadForRequest:(NSMutableDictionary<NSString *, id> *_Nonnull)request
 @property (nonatomic, assign) RNCWebViewPermissionGrantType mediaCapturePermissionGrantType;
 #endif
 
+@property (nonatomic, copy) NSString * _Nullable webViewKey;
+
 + (void)setClientAuthenticationCredential:(nullable NSURLCredential*)credential;
 + (void)setCustomCertificatesForHost:(nullable NSDictionary *)certificates;
++ (NSMutableDictionary<NSString *, id> *)baseEventWithWebView: (WKWebView *)webView;
++ (NSMutableDictionary<NSString *, id>*)createEventFromMessage:(WKScriptMessage *_Nonnull)message withMessageBodyKey: (NSString *_Nonnull)messageBodyKey withWebView: (WKWebView *)webView;
 - (void)postMessage:(NSString *_Nullable)message;
 - (void)injectJavaScript:(NSString *_Nullable)script;
 - (void)goForward;
 - (void)goBack;
 - (void)reload;
 - (void)stopLoading;
 - (void)requestFocus;
+- (void)releaseWebView;
 #if !TARGET_OS_OSX
 - (void)addPullToRefreshControl;
 - (void)pullToRefresh:(UIRefreshControl *_Nonnull)refreshControl;
+- (void)cleanUpWebView;
```

Comment:
> What's the expected MacOS support?

We're expecting zero MacOS support for the new APIs around reusing the WebView instance and sending messages to and from the WebView without a React tag. We expect existing APIs, including `removeFromSuperView`, to continue to work on MacOS.

> Why is cleanUpWebView (which already appears to check TARGET_OS_OSX in its implementation) in the !TARGET_OS_OSX block but releaseWebView (and others above) is not?

For context, you can reference our first iOS PR in our fork of react-native-webview: https://github.com/discord/react-native-webview/pull/1/files#diff-fee21637a62b4cc5dd3ff9992add07b843c999a733b03082d32cf35ef2f42bc3R529 . `removeFromSuperView` already had a `TARGET_OS_OSX` check. I pulled part of `removeFromSuperView` out into `cleanUpWebView` so that (a) `removeFromSuperView` could behave the same as before by calling `cleanUpWebView` when we're not reusing the WebView instance, and (b) we could separately call `cleanUpWebView` from `releaseWebView` when we're reusing the WebView instance.

Reply:
Your reasoning for refactoring `removeFromSuperView`/`cleanUpWebView` makes sense. However, you say that you expect `removeFromSuperView` "to continue to work on MacOS", but it calls `cleanUpWebView`, which here in the `.h` is in the `#if !TARGET_OS_OSX` block. Is that not a problem?

Reply:
Ah that's a good point. `cleanUpWebView` should be safe to call whether `TARGET_OS_OSX` is true or not. In that case, I think we should move `cleanUpWebView` in `RNCWebView.h` to outside of the `#if !TARGET_OS_OSX` block. We can update that

Reply:
Addressed here: https://github.com/react-native-webview/react-native-webview/pull/2469/commits/6a1ef637cbc96b27b1c68dd4e9c9813fbb20cb6f